docs/network-policy.md

```diff
@@ -0,0 +1,122 @@
+A lower priority value indicates higher precedence. Priority values can range
+from 1-10000.
+**Note**: Policies with same priorities will be evaluated
+indeterministically. Users should therefore take care to use priorities to
+ensure the behavior they expect.
+
+**ingress**: Each ClusterNetworkPolicy may consist of zero or more ordered
+set of ingress rules. Each rule, depending on the `action` field of the rule,
+allows or drops traffic which matches both the `from` and `ports` sections.
+The example policy contains a single rule, which allows matched traffic on a
+single port, from one of the two sources, first specified by the `podSelector`
+and second, specified by the combination of `podSelector` and
+`namespaceSelector`.
+**Note**: The order in which the ingress rules are set matter. i.e. rules will be
+evaluated in the order in which they are written.
+
+**egress**: Each ClusterNetworkPolicy may consist of zero or more ordered set of
+egress rules. Each rule, depending on the `action` field of the rule, allows
+or drops traffic which matches both the `to` and `ports` sections. The example
+policy contains a single rule, which drops matched traffic on a single port,
+to the 10.0.10.0/24 subnet specified by the `ipBlock` field.
+**Note**: The order in which the egress rules are set matter. i.e. rules will be
+evaluated in the order in which they are written.
+
+## Behavior of `to` and `from` selectors
+
+There are four kinds of selectors that can be specified in an ingress from
+section or egress to section:
+
+**podSelector**: This selects particular Pods from all Namespaces as "sources",
+if set in "ingress" section, or as "destinations", if set in "egress" section.
+
+**namespaceSelector**: This selects particular Namespaces for which all Pods are
+grouped as ingress "sources" or egress "destinations".
+
+**podSelector** and **namespaceSelector**:  A single to/from entry that specifies
+both namespaceSelector and podSelector selects particular Pods within
+particular Namespaces. 
+
+**ipBlock**: This selects particular IP CIDR ranges to allow as ingress sources
+or egress destinations. These should be cluster-external IPs, since Pod IPs are
+ephemeral and unpredictable.
+
+## Key differences from K8s NetworkPolicy
+
+- ClusterNetworkPolicy is of cluster scope, hence a `podSelector` without any
+  `namespaceSelector` selects Pods from all Namespaces.
+- There is no automatic isolation of Pods on being selected in appliedTo.
+- Ingress/Egress rules in ClusterNetworkPolicy has an `action` field which
+  specifies whether the matched rule allows or drops the traffic.
+- IPBlock field in the ClusterNetworkPolicy rules do not have the `except`
+  field. A higher priority rule can be written to deny the specific CIDR range
+  to simulate the behavior of IPBlock field with `cidr` and `except` set.
+- Rules assume the priority in which they are written. i.e. rule set at top
+  takes precedence over a rule set below it.
```

Comment:
Do we want to mention the priority hierarchy in some way? i.e. rules created for a CNP with priority 1 will all have a higher global precedence over rules created for a CNP with priority 2. I feel like making this clear can eliminate questions like "Is the second ingress rule of CNP priority1 the same priority of first rule of CNP priority2?"

Reply:
added a sub section